### PR TITLE
Genbank parser dev

### LINKED ISSUE #45: Taxonomy identifiers for Cov reference database
It would be useful to include taxonomy information in pan-genome reference to enable analyses such as taxonomy-aware summaries of bowtie2 hits. Taxonomy descriptions are included in the full FASTA deflines, but not the taxonomy id (i.e., the integer accession in the NCBI Taxonomy database). The integer accession is better because it places the sequence in a tree structure. For example, MH726362.1 is "Porcine epidemic diarrhea virus isolate GDS05, complete genome" and the taxonomy id in the Genbank record is 28295. To implement this, we need a script which can take a long list (~33k) of sequence accessions (MH726362.1) and fetch their taxonomy identifiers. I'm guessing this can be done by a simply query using Entrez or something like that.

Even better would be to have the taxonomy identifier (ideal) or species name (good) of the host as well. This can be done by downloading the Genbank record and extracting the "/host" annotation which gives the species name, e.g. for MH726362.1 you would find /host="Sus scrofa". The taxonomy identifier of the host is not given, so an additional lookup (Entrez?) would be required to retrieve it.

### PR DESCRIPTION
**Pull request by Bdegraaf1234**
The genbankParser is designed to be run as a standalone script to generate a formatted and cleaned csv table of the covid pan-genome from genbank input. It deals with most hostTaxonId mapping errors (which are plentiful), and attempts to infer these hostTaxonIds for duplicate and highly homologous entries by checking if clusters/duplicate all provide the same hostTaxonId and if so inferring it for those where none was provided (in a new column).  This infers about 1500 hostTaxonIds.

### CODE REVIEW

**Comment by rcedgar** [comment]
Didn't test, but functionality looks great, sorry again for treading on this earlier!

**Comment by rcedgar** [comment]
Minor suggestion for possible future enhancement, much of the code is essentially a lookup table which would be easier to maintain as an external file in (say) tsv format, e.g.

```
 else if (grepl(fixed = FALSE, "Vespadelus baverstocki", NoParenth, ignore.case = T)) {
    return("unclassified Scotoecus") 

```

**Comment by Bdegraaf1234** [comment]
I fully agree, I'll try and implement it somewhere in the next few days.

**Review by rcedgar — APPROVED** [review]

**Comment by taltman** [comment]
What is the status of this pull request? Are we waiting on a review from @ababaian ?

**Comment by ababaian** [comment]
It's good, it's definetly working but there are some features that need to be added, see the taxonomy issue.

**Comment by taltman** [comment]
Two questions:
* Is @Bdegraaf1234 still maintaining this code?
* Where is this code in the repo?

**Comment by taltman** [comment]
Nevermind the "where" part of the question, looked at the commits.

**Comment by ababaian** [comment]
Probably not. We can either merge and close this and have someone pick up from here.

**Comment by taltman** [comment]
I can take a crack at the metadata file, but it sounds like @r1cedgar might be doing some of this with #101 . So we should coordinate.

**Comment by rcedgar** [comment]
Parsing genbank (this issue) and uniform annotation of reference and predicted genomes (#101) are separate issues. We want both in parallel.

**Pull request closed without merging**
